src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -247,22 +247,19 @@ public string ReadString(uint codeUnitCount)
 			int length = (int)codeUnitCount;
 			VerifyRead(length);
 
-			if (!(_buffer is Buffer buffer))
-			{
-				throw new NotSupportedException("This type of buffer is not supported.");
-			}
+			var data = _buffer.GetSegment();
 
 			string result;
 			switch (UnicodeEncoding)
 			{
 				case UnicodeEncoding.Utf8:
-					result = Encoding.UTF8.GetString(buffer.Data, _bufferPosition, length);
+					result = Encoding.UTF8.GetString(data.Array!, data.Offset + _bufferPosition, length);
```

Comment:
Because the `Memory` managed by the `Buffer` is always created from a `byte[]`, so this cannot be `null. But you right for safety I've added a null check directly in the `_buffer.GetSegment()`. If the array is `null` the `GetSegment` will throw.